File: Assets/Levels/Plateformer/moveRightAction.cs
using UnityEngine;
using System.Collections;

public class moveRightAction : AbstractAction {
	
	public moveRightAction(GameObject pPlayer) : base(pPlayer){
		
	}

	public override bool execute(int duration) {
		if(curTime < duration){
			player.transform.position += new Vector3(0.05f, 0, 0);
			curTime++;
			return false;
		}
		else{
			curTime = 0;
			return true;
		}
	}
	
}

File: Assets/Levels/Plateformer/OldMovementSystem/moveRightAction.js
#pragma strict

public class moveRightAction extends AbstractAction {

	public function moveRightAction(pPlayer : GameObject){
		super(pPlayer);
	}

	public function execute(duration : int) : boolean{
		while(curTime < duration){
			player.transform.position.x += 0.05;
			curTime++;
			return false;
		}
		curTime = 0;
		return true;
		
	}
}